### DISCUSSION TITLE
How to set impulse limits for constraints

### DISCUSSION THREAD

**Discussion by 123uiui** [NVIDIA-Omniverse/PhysX#345] [Q&A]
I want to set impulse limits, but I don't know how to do it. Do I need to modify the source code?

**Comment by preist-nvidia** [comment]
You can set velocity limits on rigid bodies: https://nvidia-omniverse.github.io/PhysX/physx/5.4.2/_api_build/class_px_rigid_body.html#_CPPv4N11PxRigidBody20setMaxLinearVelocityE6PxReal

**Reply by 123uiui** [reply]
We are currently using Bullet, where we can modify the upper limit in the plimit section of the source code. As we plan to migrate to PhysX, how can we achieve this requirement?

**Reply by preist-nvidia** [reply]
Can you please be specific about what you want to limit? Solver impulse limits? Rigid body impulse limits?

**Reply by 123uiui** [reply]
Apologies for not explaining this clearly earlier! What we actually need to adjust is the Solver impulse limits.

**Reply by 123uiui** [reply]
I found the function related to impulse limits for joints in the DyCpuGpuArticulation file, specifically the computeLimitImpulse function. What we need is related to "contacts solver"

**Reply by preist-nvidia** [reply]
I think this will fit then: https://nvidia-omniverse.github.io/PhysX/physx/5.5.0/_api_build/classPxRigidBody.html#_CPPv4N11PxRigidBody27setMaxDepenetrationVelocityE6PxReal
Also works on articualtion links.

**Reply by vreutskyy** [reply]
To set contact's max impulse, use [Contact Modification](https://nvidia-omniverse.github.io/PhysX/physx/5.5.0/docs/AdvancedCollisionDetection.html#contact-modification) and [PxContactSet::setMaxImpulse(...)](https://nvidia-omniverse.github.io/PhysX/physx/5.5.0/_api_build/classPxContactSet.html#_CPPv4N12PxContactSet13setMaxImpulseE5PxU326PxReal) function.
